### PR TITLE
Extend list offset API to return next local offset

### PR DESCRIPTION
**Pull request by ying-zheng**

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/requests/ListOffsetRequest.java [hunk 1]
```diff
@@ -39,6 +39,7 @@
 import static org.apache.kafka.common.protocol.CommonFields.TOPIC_NAME;
 
 public class ListOffsetRequest extends AbstractRequest {
+    public static final long NEXT_LOCAL_TIMESTAMP = -3L;
```
Comment (satishd) [inline clients/src/main/java/org/apache/kafka/common/requests/ListOffsetRequest.java:42]: `EARLIEST_LOCAL_TIMESTAMP` looks to be a better name.

**Review by satishd — COMMENTED** [review]
thanks @ying-zheng for the PR, left a minor comment and the below questions.

Are not the below changes needed?

- `FetchRequest.json` and `FetchResponse.json` should have valid versions as `0-11` instead of `0-12`and `FetchResponse.json` should remove the below field.
```
 {"name":  "NextLocalOffset", "type": "int64", "versions": "12+", "default": "-1", "ignorable": true,
          "about":  "Next local offset available on leader when tiered storage is enabled for a topic."},
```

- `ListOffsetRequest.json` and `ListOffsetResponse.json` should bump valid versions as `0-6`

File: clients/src/main/java/org/apache/kafka/common/requests/ListOffsetRequest.java (hunk 1 above)
Comment (ying-zheng) [reply clients/src/main/java/org/apache/kafka/common/requests/ListOffsetRequest.java:42]: The offset we are looking for is "the earliest offset that has not been sent to remote storage". It's not the earliest offset in local storage. After a segment is sent to remote storage, we will still keep it in local storage for a pretty long time

Comment (satishd) [reply clients/src/main/java/org/apache/kafka/common/requests/ListOffsetRequest.java:42]: I understand what this offset is for :) It looks like neither of the names are intuitive. We can address this later.

**Comment by satishd** [comment]
thanks @ying-zheng for addressing the review comments, LGTM

**Pull request merged**